elixir/apps/domain/lib/domain/config/definitions.ex

```diff
@@ -590,7 +590,20 @@ defmodule Domain.Config.Definitions do
     # TODO: validate opts are present if adapter is not NOOP one
     default: %{},
     sensitive: true,
-    dump: &Dumper.keyword/1
+    dump: fn map ->
+      keyword = Dumper.keyword(map)
+
+      cond do
+        compile_config!(:outbound_email_adapter) == Swoosh.Adapters.SMTP ->
```

Comment:
@Intuinewin I think we don't need to check for the adapter and can assume that TLS options should be always mapped, like so:

```
dump: fn value ->
      {tls_opts, value} = Map.pop(value, "tls_opts", %{})

      Dumper.keyword(value)
      |> Keyword.put(:tls_opts, Dumper.dump_ssl_opts(tls_opts))
    end
```

Reply:
Is it better?

Reply:
Yes, thank you